third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -5259,11 +5321,22 @@ void OnNewRootFrame(v8::Isolate* isolate, LocalFrame* localFrame) {
   }
 }
 
-void OnNewWindow2(v8::Isolate* isolate, LocalFrame* localFrame) {
+void OnNewWindow2(v8::Isolate* isolate, LocalFrame* localFrame, v8::Local<v8::Context> newContext) {
   recordreplay::AutoMarkReplayCode amrc;
   v8::Local<v8::Context> context = isolate->GetCurrentContext();
   RunScript(isolate, context, gOnNewWindowScript,
             "record-replay-devtools-OnNewWindow");
+
```

Comment:
Can we get a new root frame / context on a new window?  I was expecting to see another `RecordReplaySetDefaultContext` here.

Reply:
* This is just another event in a line of events during `LocalWindowProxy::Initialize`.
* `RecordReplaySetDefaultContext` was already called in `OnNewRootFrame` (if this new window had `IsOutermostMainFrame` = `true`).
* We are not concerned with *all contexts*, but only with *root contexts*: The problem we are fixing here is about (1) `contextGroupId`s being incorrect and (2) default contexts disappearing:
  1. `contextGroupId` are pegged to their root
* We thus don't care about individual child contexts.

Reply:
That being said... I just realized another problem I need to fix 😭